src/java/org/apache/cassandra/security/PEMBasedSslContextFactory.java

```diff
@@ -92,33 +92,56 @@ public final class PEMBasedSslContextFactory extends FileBasedSslContextFactory
     private static final Logger logger = LoggerFactory.getLogger(PEMBasedSslContextFactory.class);
     private String pemEncodedKey;
     private String keyPassword;
+    private String pemEncodedOutboundKey;
+    private String outboundKeyPassword;
     private String pemEncodedCertificates;
     private boolean maybeFileBasedPrivateKey;
+    private boolean maybeFileBasedOutboundPrivateKey;
     private boolean maybeFileBasedTrustedCertificates;
 
     public PEMBasedSslContextFactory()
     {
     }
 
-    public PEMBasedSslContextFactory(Map<String, Object> parameters)
+    private void validatePassword(boolean outbound)
```

Comment:
The method does more than password validation. 
In the `if (StringUtils.isEmpty(password))` branch, it initializes the password. It should be moved to the contractor and call `StringUtils.defaultString` to have the same initialization fallback. 

Then you can have a simple if-else statement in the body and remove the `outbound` parameter for the method. 

For example
```java
    // call it once finishing initialize both inbound and outbound.
    private void validatePassword()
    {
        if (!keystoreContext.passwordMatchesIfPresent(keyPassword) 
            || !outboundKeystoreContext.passwordMatchesIfPresent(outboundKeyPassword))
        {
            // throw exception
        }
        else
        {
            // log warn
        }
    }
```